service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/realm/UserPassCredentialsMatcher.java

```diff
@@ -179,3 +199,42 @@ public boolean doCredentialsMatch(AuthenticationToken authenticationToken, Authe
     }
 
 }
+
+interface PasswordChecker {
+
+    boolean checkPassword(String tokenPassword, Credential infoCredential);
+
+}
+
+class CachePasswordChecker implements PasswordChecker {
```

Comment:
It would be better to have a password cacher which is independent from the actual password checker implementation - e.g.: a wrapper/decorator around another PasswordChecker. That we we can compose different strategies, and/or test independently the caching algorithm and the password checking one.